frontend/src/home/Home.tsx

```diff
@@ -186,16 +186,17 @@ class HomeComponent extends React.Component<HomeProps, HomeState> {
   }
 
   updateWarnings(newState: HomeState) {
-    const warnings = produceWarnings(newState.schedule);
+    const container = produceWarnings(newState.schedule);
+    const normal = container.normalWarnings;
```

Comment:
@NilsBacke, this `container` variable here is an object, with 2 fields: normalWarnings, and courseWarnings. The prior are normal warnings, and I changed this code to handle them like normal (hah!). The `container.courseWarnings` field contains a list of `CourseWarning`, each of which is a normal warning with the added properties of `subject` and `classId`. 

Hopefully this is what you need!